aibolit/metrics/cognitiveC/cognitive_c.py

```diff
@@ -1,130 +1,139 @@
+        if operator not in logical_operators:
+            return []
 
-                self._increment_logical_operators(each_block, each_block.operator)
+        left_sequence = self._create_logical_operators_sequence(ast, left_side_node)
+        right_sequence = self._create_logical_operators_sequence(ast, right_side_node)
+        return left_sequence + [operator] + right_sequence
 
-    def _is_recursion_call(self, block: MethodInvocation) -> bool:
-        if self.method_name == block.member:
-            return True
-        return False
+    def _is_recursion_call(self, ast, node) -> bool:   # type: ignore
+        if ast.get_type(node) == ASTNodeType.METHOD_INVOCATION:
+            if self.method_name == self._get_node_name(ast, node):
+                return True
+            return False
 
-    def _nested_methods(self, block: MethodDeclaration, nested_level: int) -> None:
+    def _nested_methods(self, ast, node, nested_level: int) -> None:
         original_name = self.method_name
-        self.method_name = block.name
-        self._get_complexity(block, nested_level + 1)
+        self.method_name = self._get_node_name(ast, node)
+        self._get_complexity(ast, node, nested_level + 1)
         self.method_name = original_name
 
-    def _get_complexity(self, block: Any, nested_level: int) -> None:
-        block_arr = block if isinstance(block, List) else [block]
-
-        for each_block in block_arr:
-            if hasattr(each_block, 'children'):
-
-                if type(each_block) == MethodDeclaration and each_block.name != self.method_name:
-                    self._nested_methods(each_block, nested_level)
+    def _get_complexity(self, ast: Any, each_block: int, nested_level: int) -> None:
+        each_block_name = self._get_node_name(ast, each_block)
+        each_block_type = ast.get_type(each_block)
 
-                elif isinstance(each_block, IfStatement):
-                    self._check_if_statement(each_block, nested_level)
+        if each_block_type == ASTNodeType.METHOD_DECLARATION and each_block_name != self.method_name:
+            self._nested_methods(ast, each_block, nested_level)
 
-                elif type(each_block) in increment_for and type(each_block) in nested_for:
-                    self.complexity += 1 + nested_level
-                    self._traverse_childs(each_block, nested_level + 1)
+        elif each_block_type == ASTNodeType.IF_STATEMENT:
+            self._check_if_statement(ast, each_block, nested_level)
 
-                elif type(each_block) in increment_for and type(each_block) not in nested_for:
-                    if isinstance(each_block, BinaryOperation):
-                        if each_block.operator in logical_operators:
-                            self.complexity += 1
-                            self._increment_logical_operators(each_block, each_block.operator)
+        elif each_block_type in increment_for and each_block_type in nested_for:
```

Comment:
we better divide `increment_for` set into two groups: `nested_for` and `not_nested_for`. This would simplify this and the next `if` condition.

Reply:
I decided not to change it